builtin/logical/pki/path_acme_new_account.go

```diff
@@ -285,3 +335,52 @@ func (b *backend) acmeNewAccountCreateHandler(acmeCtx *acmeContext, r *logical.R
 	resp.Data[logical.HTTPStatusCode] = http.StatusCreated
 	return resp, nil
 }
+
+func (b *backend) acmeNewAccountUpdateHandler(acmeCtx *acmeContext, userCtx *jwsCtx, contact []string, status string) (*logical.Response, error) {
+	if !userCtx.Existing {
+		return nil, fmt.Errorf("cannot submit to account updates without a 'kid': %w", ErrMalformed)
+	}
+
+	if !b.acmeState.DoesAccountExist(acmeCtx, userCtx.Kid) {
+		return nil, fmt.Errorf("an account with this key does not exist: %w", ErrAccountDoesNotExist)
+	}
+
+	account, err := b.acmeState.LoadAccount(acmeCtx, userCtx.Kid)
+	if err != nil {
+		return nil, fmt.Errorf("error loading account: %w", err)
+	}
+
+	// Per RFC 8555 7.3.6 Account deactivation, if we were previously deactivated, we should return
+	// unauthorized. There is no way to reactivate any accounts per ACME RFC.
+	if account.Status != StatusValid {
+		// Treating "revoked" and "deactivated" as the same here.
+		return nil, ErrUnauthorized
+	}
+
+	shouldUpdate := false
+	// Check to see if we should update, we don't really care about ordering
+	if !strutil.EquivalentSlices(account.Contact, contact) {
+		shouldUpdate = true
+		account.Contact = contact
+	}
+
+	// Check to process account de-activation status was requested.
+	// 7.3.6. Account Deactivation
+	if string(StatusDeactivated) == status {
+		shouldUpdate = true
+		// TODO: This should cancel any ongoing operations (do not revoke certs),
+		//       perhaps we should delete this account here?
```

Comment:
I think we'll persist the account until the tidy cleans it up so that any pending operations have time to die gracefully.